static/js/subs.js

```diff
@@ -663,6 +664,45 @@ exports.remove_user_from_stream = function (user_email, stream_name, success, fa
     });
 };
 
+exports.filter_streams = function (search_list, stream_ids) {
+    if (search_list === 0) {
+        return;
+    }
+    var dom = search_list.attr('class');
+    var search_term = search_list.expectOne().val().trim();
+    var search_terms = search_term.toLowerCase().split(",");
+    search_terms = _.map(search_terms, function (s) {
+        return s.trim();
+    });
+
+    _.each(stream_ids, function (stream_id) {
+        var stream_name = stream_data.get_sub_by_id(stream_id).name;
+        stream_name = stream_name.toLowerCase().split(/\s+/);
+        stream_name = _.map(stream_name, function (s) {
+            return s.trim();
+        });
+
+        _.any(search_terms, function (search_term) {
```

Comment:
Please rewrite this in a functional style. Calls to `_.any` should not have side effects